src/Compilers/CSharp/Portable/Parser/LanguageParser.cs

```diff
@@ -8879,6 +8892,11 @@ private ArgumentSyntax ParseArgumentExpression(bool isIndexer)
                 expression = this.ParseSubExpression(0);
             }
 
+            if (refOrOutKeyword != null && IsLambda(expression.Kind))
+            {
+                expression = this.AddError(expression, ErrorCode.ERR_RefLvalueExpected);
```

Comment:
It is consistent with the legacy compiler. The legacy compiler reports `CS1660` as well.